zio-kafka/src/main/scala/zio/kafka/consumer/Consumer.scala

```diff
@@ -488,27 +598,135 @@ private[consumer] final class ConsumerLive private[consumer] (
 
               tp -> stream
           }
-        }
-    }
+        },
+      streamControl.stop
+    )
+  }
+
+  override def partitionedAssignmentStream[R, K, V](
+    subscription: Subscription,
+    keyDeserializer: Deserializer[R, K],
+    valueDeserializer: Deserializer[R, V]
+  ): Stream[
+    Throwable,
+    Chunk[
+      (
+        TopicPartition,
+        ZStream[R, Throwable, CommittableRecord[K, V]]
+      )
+    ]
+  ] = ZStream.unwrapScoped {
+    for {
+      streamControl <- partitionedAssignmentStreamWithControl(subscription, keyDeserializer, valueDeserializer)
+    } yield streamControl.stream
   }
 
+  override def partitionedAssignmentStreamWithGracefulShutdown[R, K, V](
+    subscription: Subscription,
+    keyDeserializer: Deserializer[R, K],
+    valueDeserializer: Deserializer[R, V],
+    shutdownTimeout: Duration = 15.seconds
+  )(
+    withStream: Stream[Throwable, Chunk[(TopicPartition, ZStream[R, Throwable, CommittableRecord[K, V]])]] => ZIO[
+      R,
+      Throwable,
+      Any
+    ]
+  ): ZIO[R, Throwable, Any] = runWithGracefulShutdown[Stream[Throwable, Chunk[
+    (TopicPartition, ZStream[R, Throwable, CommittableRecord[K, V]])
+  ]], R, Throwable](
+    partitionedAssignmentStreamWithControl(subscription, keyDeserializer, valueDeserializer),
+    shutdownTimeout
+  )(
+    withStream
+  )
+
   override def partitionedStream[R, K, V](
     subscription: Subscription,
     keyDeserializer: Deserializer[R, K],
     valueDeserializer: Deserializer[R, V]
-  ): ZStream[Any, Throwable, (TopicPartition, ZStream[R, Throwable, CommittableRecord[K, V]])] =
-    partitionedAssignmentStream(subscription, keyDeserializer, valueDeserializer).flattenChunks
+  ): Stream[
+    Throwable,
+    (
+      TopicPartition,
+      ZStream[R, Throwable, CommittableRecord[K, V]]
+    )
+  ] = ZStream.unwrapScoped {
+    for {
+      streamControl <- partitionedAssignmentStreamWithControl(subscription, keyDeserializer, valueDeserializer)
+    } yield streamControl.stream.flattenChunks
+  }
+
+  override def partitionedStreamWithGracefulShutdown[R, K, V](
+    subscription: Subscription,
+    keyDeserializer: Deserializer[R, K],
+    valueDeserializer: Deserializer[R, V],
+    shutdownTimeout: Duration = 15.seconds
+  )(
+    withStream: Stream[Throwable, (TopicPartition, ZStream[R, Throwable, CommittableRecord[K, V]])] => ZIO[
+      R,
+      Throwable,
+      Any
+    ]
+  ): ZIO[R, Throwable, Any] =
+    partitionedAssignmentStreamWithGracefulShutdown(subscription, keyDeserializer, valueDeserializer, shutdownTimeout) {
+      stream =>
+        withStream(stream.flattenChunks)
+    }
 
   override def plainStream[R, K, V](
     subscription: Subscription,
     keyDeserializer: Deserializer[R, K],
     valueDeserializer: Deserializer[R, V],
-    bufferSize: Int
+    bufferSize: RuntimeFlags
```

Comment:
This is probably an IDE mistake
```suggestion
    bufferSize: Int
```